pkg/userdata/ubuntu/provider.go

```diff
@@ -265,6 +271,13 @@ write_files:
     {{- if .OSConfig.DistUpgradeOnBoot }}
     DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" dist-upgrade -y
     {{- end }}
+
+    # Update grub to include kernel command options to enable swap accounting.
+    if grep -v -q swapaccount=1 /proc/cmdline
+    then
+      update-grub
+      touch /var/run/reboot-required
+    fi
     if [[ -e /var/run/reboot-required ]]; then
```

Comment:
this section need to be deleted

Reply:
Removing this and using the power_state module instead is causing a race condition in one of the test cases. Basically power_state triggers a reboot _during_ the OS update process, rendering apt in a broken state that requires a manual fix and causes the node to be stuck in a loop. 

I removed the powerstate module entry and restored this to avoid this effect.